src/main/java/com/amplitude/Amplitude.java

```diff
@@ -81,8 +83,8 @@ private void updateHttpCall(HttpCallMode updatedHttpCallMode) {
       httpCallMode = updatedHttpCallMode;
       httpCall =
           (updatedHttpCallMode == HttpCallMode.BATCH_HTTPCALL)
-              ? new BatchHttpCall(apiKey)
-              : new GeneralHttpCall(apiKey);
+              ? new BatchHttpCall(apiKey, logger)
```

Comment:
Not a fan to pass the logger with it, normal way to get a logger is like `Logger LOGGER = LogManager.getLogger();`, we can see if we want to adjust `AmplitudeLog` to do similar like getInstance